src/main/proxy.ts

```diff
@@ -130,7 +130,8 @@ export class LensProxy {
     if (req.url.startsWith("/api-kube/")) {
       delete req.headers.authorization
       req.url = req.url.replace("/api-kube", "")
-      return await contextHandler.getApiTarget()
+      const timeout = req.url.includes("watch=") ? 0 : 30000
```

Comment:
I hope this naive check is enough here.

Reply:
We might be able to check request type from headers (watches are event streams).